### DISCUSSION TITLE
files.treatAs

### DISCUSSION THREAD

**Discussion by jcayzac** [biomejs/biome#3355] [Ideas]
I don't think it's possible currently to use Biome on files whose extensions it does not understand.

For example, `.vscode/*.code-snippets` are JSON files, but Biome doesn't understand that and doesn't apply JSON rules to those.

I'm thinking that it would be nice if we could just add any pattern we want to be recognized as a certain file type, e.g.

```json
{
  "files": {
    "treatAs": {
      "json": "**/*.code-snippets"
    }
  }
}
```

**Comment by ematipico** [comment]
I like it! I would change the API a bit but the code idea feels useful.

Also, we could use git for that too.
